### PR TITLE
add support for bigknob

### PR DESCRIPTION
**Pull request by LeafCutterLabs**
Add support for bigknob macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ X] I have updated the documentation accordingly.
- [ X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ X] I have added tests to cover my changes.
- [ X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/leafcutterlabs/bigknob/config.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ *
+*/
+
+// Srating with closest to USB port 
+#define DIRECT_PINS {   \
+    { B7, D4, D6, F6, F7} \
+}
+//speed for double tap
+#define TAPPING_TERM 300
+
```
Comment (zvecr) [inline keyboards/leafcutterlabs/bigknob/config.h:52]:
```suggestion
```

File: keyboards/leafcutterlabs/bigknob/rules.mk [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+TAP_DANCE_ENABLE = yes      # Double press does something different
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/leafcutterlabs/bigknob/rules.mk:27]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes        # Enable rotary encoders
```

File: keyboards/leafcutterlabs/bigknob/config.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xCEEB
+#define PRODUCT_ID      0x0007
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    leafcutterlabs
+#define PRODUCT         bigKNOB
+#define DESCRIPTION     5 key with rotary encoder board
+
```
Comment (zvecr) [inline keyboards/leafcutterlabs/bigknob/config.h:29]:
```suggestion

```

File: keyboards/leafcutterlabs/bigknob/bigknob.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include "bigknob.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (zvecr) [inline keyboards/leafcutterlabs/bigknob/bigknob.c:5]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, all `*.h` and `*.c` files need a GPL compatible header added.

File: keyboards/leafcutterlabs/bigknob/config.h [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+// Srating with closest to USB port 
+#define DIRECT_PINS {   \
+    { B7, D4, D6, F6, F7} \
+}
+#define UNUSED_PINS
+
+/* rotary encoder 1,2,3 closest to usb port is 0*/
+#define ENCODERS_PAD_A { D0}
+#define ENCODERS_PAD_B { D2}
+#define ENCODER_RESOLUTION 8 //default/suggested
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigknob/config.h:53]: The default resolution is 4. Are you sure this works?

File: keyboards/leafcutterlabs/bigknob/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+enum {
+  TD_RGB = 0
+};
+
+void dance_rgb_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+    register_code (KC_MEDIA_NEXT_TRACK);
+	unregister_code (KC_MEDIA_NEXT_TRACK);
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigknob/keymaps/default/keymap.c:22]:
```suggestion
    tap_code(KC_MNXT);
```

File: keyboards/leafcutterlabs/bigknob/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+![](https://github.com/LeafCutterLabs/bigKNOB/blob/master/Media/thinkpad.jpg)
+
+* Keyboard Maintainer: [LeafCutterLabs](https://github.com/LeafCutterLabs)
+* Hardware Supported: Atmega32u4
+* Hardware availability: Case files are available [here](https://github.com/LeafCutterLabs/bigKNOB)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make bigknob:default
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigknob/readme.md:13]:
```suggestion
    make leafcutterlabs/bigknob:default
```

File: keyboards/leafcutterlabs/bigknob/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# bigknob
+
+A 4 key Macropad with Rotatry Encoder based on Atmega32u4.
+
+![](https://github.com/LeafCutterLabs/bigKNOB/blob/master/Media/thinkpad.jpg)
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigknob/readme.md:5]: Please rehost this on imgur and add `l` to the end of the filename to make it thumbnail sized.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/leafcutterlabs/bigknob/config.h (hunk 5 above)
Comment (LeafCutterLabs) [reply keyboards/leafcutterlabs/bigknob/config.h:53]: Depends on the encoder, but 4 would be better for a more general approach

Comment (fauxpark) [reply keyboards/leafcutterlabs/bigknob/config.h]: I've not seen an encoder which needed a resolution of 8 - this is the number of pulses between each detent, and can be worked out from the quadrature table in the datasheet. Usually it is either 2 or 4.

Comment (LeafCutterLabs) [reply keyboards/leafcutterlabs/bigknob/config.h]: Changed to 4

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/leafcutterlabs/bigknob/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# bigknob
+
+A 4 key Macropad with Rotatry Encoder based on Atmega32u4.
```
Comment (fauxpark) [inline keyboards/leafcutterlabs/bigknob/readme.md:3]:
```suggestion
A 4 key Macropad with Rotary Encoder based on Atmega32u4.
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by LeafCutterLabs — COMMENTED** [review]
corrected spelling

**Review by LeafCutterLabs — COMMENTED** [review]
corrected spelling

**Review by fauxpark — APPROVED** [review]

**Pull request merged**